samples/cpp/qrcode.cpp

```diff
@@ -89,41 +98,79 @@ int liveQRCodeDetect(const string& out_file)
         return -4;
     }
 
+    cout << "Press 'm' to switch between detectAndDecode and detectAndDecodeMulti" << endl;
     QRCodeDetector qrcode;
     TickMeter total;
+    bool switch_mode = false;
+    char c = (char)waitKey(30);
     for(;;)
     {
-        Mat frame, src, straight_barcode;
-        string decode_info;
-        vector<Point> transform;
+        Mat frame, src;
         cap >> frame;
         if (frame.empty())
         {
             cout << "End of video stream" << endl;
             break;
         }
+        if (!out_origin_file.empty())
+            imwrite(out_origin_file, frame); // write original frame
```

Comment:
Saving each frame slowdowns demo significantly. Image I/O is not optimized (because it is for debug purposes here).

Original and processed frames should be saved by pressing key only (it was space ' '). Both at once.